### PR TITLE
Corrected create rule method, seperate method for each object

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
Corrected Create rule method , Seperate method for each object
**Which issue(s) this PR fixes** (optional)
Closes #https://portworx.atlassian.net/browse/PA-494

**Special notes for your reviewer**:

### CODE REVIEW

File: drivers/backup/backup.go [hunk 1]
```diff
@@ -323,7 +323,7 @@ type Rule interface {
 	UpdateOwnershipRule(ctx context.Context, req *api.RuleOwnershipUpdateRequest) (*api.RuleOwnershipUpdateResponse, error)
 
 	// CreateRuleForBackup creates backup rule
-	CreateRuleForBackup(ruleName string, orgID string, appList []string, prePostFlag string, ps map[string]map[string]string) (bool, string, error)
+	CreateRuleForBackup(appName string, orgID string, prePostFlag string) (bool, error, string)
```
Comment (lsrinivas-pure) [inline drivers/backup/backup.go:326]: make error as last return object

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -298,19 +287,20 @@ func TeardownForTestcase(contexts []*scheduler.Context, providers []string, Clou
 	}
 	log.InfoD("Deleting bucket,backup location and cloud setting")
 	for i, provider := range providers {
-		backup_location_name := fmt.Sprintf("%s-%s", "location", provider)
+		//backup_location_name := fmt.Sprintf("%s-%s", "location", provider)
```
Comment (kshithijiyer-px) [inline tests/backup/backup_test.go:290]: If not need can we remove it?

File: drivers/backup/portworx/portworx.go [hunk 3]
```diff
@@ -1098,67 +1098,82 @@ func (p *portworx) GetBackupUID(ctx context.Context, orgID, backupName string) (
...
+			"pod_selector_list": {"app=postgres"},
+			"container":         {"", ""},
+		},
+		},
 	}
 )
 
-func (p *portworx) CreateRuleForBackup(ruleName string, orgID string, appList []string, prePostFlag string,
-	ps map[string]map[string]string) (bool, string, error) {
+func (p *portworx) CreateRuleForBackup(appName string, orgID string, prePostFlag string) (bool, error, string) {
```
Comment (lsrinivas-pure) [inline drivers/backup/portworx/portworx.go:1127]: same here

File: tests/backup/backup_test.go [hunk 4]
```diff
@@ -177,45 +173,32 @@ var _ = Describe("{BasicBackupCreation}", func() {
...
-			var err error
 			log.InfoD("Creating pre rule for deployed apps")
-			preRuleStatus, preRuleUid, err = Inst().Backup.CreateRuleForBackup(PreRuleName, "default", appList,
-				"pre", ps)
-			log.FailOnError(err, "Creating pre rule for deployed apps failed")
-			dash.VerifyFatal(preRuleStatus, true, "Verifying pre rule for backup")
+			for i := 0; i < len(appList); i++ {
+				preRuleStatus, err, ruleName := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "pre")
+				preRuleNameList = append(preRuleNameList, ruleName)
+				log.FailOnError(err, "Creating pre rule for deployed apps failed")
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:184]: handle the error first and then use ruleName

File: tests/backup/backup_test.go [hunk 5]
```diff
@@ -177,45 +173,32 @@ var _ = Describe("{BasicBackupCreation}", func() {
...
+				dash.VerifyFatal(preRuleStatus, true, "Verifying pre rule for backup")
+			}
 			log.InfoD("Creating post rule for deployed apps")
-			postRuleStatus, postRuleUid, err = Inst().Backup.CreateRuleForBackup(PostRuleName, "default", appList,
-				"post", ps)
-			log.FailOnError(err, "Creating post rule for deployed apps failed")
-			dash.VerifyFatal(postRuleStatus, true, "Verifying Post rule for backup")
+			for i := 0; i < len(appList); i++ {
+				postRuleStatus, err, ruleName := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "post")
+				postRuleNameList = append(postRuleNameList, ruleName)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:190]: same here

File: tests/backup/backup_test.go [hunk 6]
```diff
@@ -252,12 +234,16 @@ var _ = Describe("{BasicBackupCreation}", func() {
...
 			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
 		})
-
 		Step("Taking backup of applications", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting context")
+			preRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, preRuleNameList[0])
+			postRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, postRuleNameList[0])
 			for _, namespace := range bkpNamespaces {
+				log.Info(" This is ns", namespace)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:243]: use infof

File: tests/backup/backup_test.go [hunk 7]
```diff
@@ -270,20 +256,23 @@ var _ = Describe("{BasicBackupCreation}", func() {
...
-		// Have added an independent function for deleting rules,policy.
-		// Likewise will split TeardownForTestcase function for each object deletion
-		err := Inst().Backup.DeleteRuleForBackup(orgID, PreRuleName)
-		dash.VerifyFatal(err, nil, "Deleting pre backup rules")
-		err = Inst().Backup.DeleteRuleForBackup(orgID, PostRuleName)
-		dash.VerifyFatal(err, nil, "Deleting post backup rules")
-		err = Inst().Backup.DeleteBackupSchedulePolicy(orgID, policyList)
-		dash.VerifyFatal(err, nil, "Deleting backup schedule policies")
+		for _, ruleName := range preRuleNameList {
+			err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:260]: add rulename if the verfication log

File: tests/backup/backup_test.go [hunk 8]
```diff
@@ -270,20 +256,23 @@ var _ = Describe("{BasicBackupCreation}", func() {
...
+			dash.VerifySafely(err, nil, "Deleting  backup rules")
+		}
+		for _, ruleName := range postRuleNameList {
+			err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
+			dash.VerifySafely(err, nil, "Deleting  backup rules")
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:265]: same here

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/backup/backup_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:290]: done

File: drivers/backup/portworx/portworx.go (hunk 3 above)
Comment (snigdha-px) [reply drivers/backup/portworx/portworx.go:1127]: done

File: tests/backup/backup_test.go (hunk 4 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:184]: done

File: tests/backup/backup_test.go (hunk 7 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:260]: done

File: tests/backup/backup_test.go (hunk 8 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:265]: done

File: drivers/backup/auth.go [hunk 9]
```diff
@@ -27,7 +27,7 @@ const (
 	// PxCentralAdminSecretName secret for PxCentralAdminUser
 	PxCentralAdminSecretName = "px-central-admin"
 	// PxCentralAdminSecretNamespace namespace of PxCentralAdminSecretName
-	PxCentralAdminSecretNamespace = "px-backup"
+	PxCentralAdminSecretNamespace = "central"
```
Comment (kshithijiyer-px) [inline drivers/backup/auth.go:30]: Why are we changing it to central?

File: drivers/backup/auth.go [hunk 10]
```diff
@@ -46,7 +46,7 @@ const (
 	// AdminTokenSecretName which has admin user jwt token information
 	AdminTokenSecretName = "px-backup-admin-secret"
 	// AdminTokenSecretNamespace which has admin user jwt token information
-	AdminTokenSecretNamespace = "px-backup"
+	AdminTokenSecretNamespace = "central"
```
Comment (kshithijiyer-px) [inline drivers/backup/auth.go:49]: Same as above

File: drivers/scheduler/k8s/specs/postgres/px-postgres-app.yaml [hunk 11]
```diff
@@ -8,7 +8,7 @@ spec:
       maxSurge: 1
       maxUnavailable: 1
     type: RollingUpdate
-  replicas: 2
+  replicas: 1
```
Comment (kshithijiyer-px) [inline drivers/scheduler/k8s/specs/postgres/px-postgres-app.yaml:11]: Are you sure you want to make this change?

File: drivers/scheduler/k8s/specs/postgres/pxd/px-storage-class.yaml [hunk 12]
```diff
@@ -5,13 +5,6 @@ metadata:
...
-  {{ if .Repl }}
-  repl: "{{ .Repl }}"
-  {{ else }}
-  repl: "3"{{ end }}
+  repl: "3"
   priority_io: "high"
-  {{ if .Fs }}
-  fs: {{ .Fs }}{{ end }}
-  {{ if .AggregationLevel }}
-  aggregation_level: "{{ .AggregationLevel }}"{{ end }}
```
Comment (kshithijiyer-px) [inline drivers/scheduler/k8s/specs/postgres/pxd/px-storage-class.yaml:16]: Are you sure you want to do remove these?

File: tests/basic/sharedv4_test.go [hunk 13]
```diff
@@ -1,5 +1,6 @@
 package tests
 
+/*
```
Comment (kshithijiyer-px) [inline tests/basic/sharedv4_test.go:3]: Do you want to comment out this/

**Review by kshithijiyer-px — COMMENTED** [review]
A few minor changes, please fix them so that we can quickly merge it.

**Review by kshithijiyer-px — APPROVED** [review]

**Pull request merged**